src/bike.cpp

```diff
@@ -5,14 +5,31 @@
 
 bike::bike() { elapsed.setType(metric::METRIC_ELAPSED); }
 
+void bike::changeResistanceRange(int8_t lower, int8_t upper) {
+    // 0.0 is floor and 1.0 is ceil
+    double lastResistanceOffsetPerc = 0.0;
+    if (RequestedLowerResistance >= 0) {
+        lastResistanceOffsetPerc = (Resistance.value() - RequestedLowerResistance) / (RequestedUpperResistance - RequestedLowerResistance);
```

Comment:
I think the drift is caused by the fact that I'm using `Resistance.value()` as it's the absolute value emitted from the bike which is most likely an `int`. Due to peloton <-> echelon conversion, `Resistance.value()` might be +/-3 in Peloton units from what `calculatedResistance` is and we lose the granularity.

To fix this, we might need to track whether or not the user changed `Resistance.value()` by turning the knob or using the UI. If the user did not change it we should use `RequestedResistance`, otherwise use `Resistance.value()`

Reply:
one of the most common request was getting the event of the knob movement, in order to change the gear tile. Maybe we can create this event and you can use somehow on this? what do you think?